draft/XXXX-configuration_handling/README.md

```diff
@@ -0,0 +1,202 @@
+---
+RFC: XXXX
+Author: Roni Choudhury
+Created: 2020-06-29
+Created: 2020-12-09
+Last-Modified: 2020-12-09
+---
+
+# RFC XXXX: Configuration Handling
+
+One of the core features of our vision for Tinker Engine is the ability to
+accept a kind of "meta-configuration" from which concrete configurations are
+generated, used to drive individual protocol runs, with results aggregated
+together in some way that enables post-hoc data analysis. This RFC lays out some
+ideas for some of the core features such a handler will have, along with an
+overall architecture for configuration hanlding in Tinker Engine.
+
+## Proposals
+
+This RFC will discuss a few concrete proposals. One is the overall architcture
+by which the Tinker Engine runtime receives, processes, and submits
+configuration data to a protocol. Another is the method by which a configuration
+file can express variation in its parameters. A third is a fallback mechanism
+allowing programmatic configuration handling when the declarative methods are
+not sufficient for some use case.
+
```

Comment:
Right. This is specifically about dealing with configuration handling. We'll need to draw up some designs for how to collect results (as it's not at all clear how it should be done, at least to me).

Reply:
Tracking config parameters saving should be linked to results since that is one of the main reasons to save them is to see the evaluation of each set of configs.  The link is that we would like to make all config objects linked to metrics so that we can fully evaluate the metrics w.r.t the metrics.  I think DVC tag line is great "Metrics are first-class citizens".

Reply:
I agree that each concrete configuration that pops out of the configuration handling routines should absolutely be stored along with the results. This aids reproducibility in general, and as you mention, it would be helpful for evaluating performance as a function of these parameters.

I'll update the text to mention the eventual saving of these concrete configurations along with the other runtime outputs.

Reply:
I would be worried if the implementation wasn't specifically setup to allow for this to happen

Reply:
See bf10e24607a486fa7e43db0943137bf08ef9f9e8.

Reply:
This is good